### PR TITLE
Add a "Open Config Location" button

### PR DESCRIPTION
**Pull request by VolcaEM**
This works just like the "Open Transferable Shader Cache" button does, and its primary use is to be able to easily find the "configuration file" where the per-game configuration for a game is saved.

screenshot

### CODE REVIEW

File: src/yuzu/main.cpp [hunk 1]
```diff
@@ -1887,6 +1893,23 @@ void GMainWindow::OnOpenFAQ() {
     OpenURL(QUrl(QStringLiteral("https://yuzu-emu.org/wiki/faq/")));
 }
 
+void GMainWindow::OpenFolderSpecifyingFile(const QString& file) {
+
```
Comment (lioncash) [inline src/yuzu/main.cpp:1897]: Unnecessary newline

File: src/yuzu/main.cpp [hunk 2]
```diff
@@ -1887,6 +1893,23 @@ void GMainWindow::OnOpenFAQ() {
...
+    // Windows supports opening a folder with selecting a specified file in explorer. On every other
+    // OS we just open the folder without preselecting the file.
+#if defined(Q_OS_WIN)
+    const QString explorer = QStringLiteral("explorer");
+    QStringList param;
+    if (!QFileInfo(file).isDir()) {
+        param << QStringLiteral("/select,");
```
Comment (lioncash) [inline src/yuzu/main.cpp:1904]:
```suggestion
        param.push_back(QStringLiteral("/select,"));
```

File: src/yuzu/main.cpp [hunk 3]
```diff
@@ -1887,6 +1893,23 @@ void GMainWindow::OnOpenFAQ() {
...
+    }
+    param << QDir::toNativeSeparators(file);
```
Comment (lioncash) [inline src/yuzu/main.cpp:1906]:
```suggestion
    param.push_back(QDir::toNativeSeparators(file));
```

**Comment by FearlessTobi** [comment]
Why do we need this exactly?
Is there a need for users to edit those config files manually?

**Comment by lat9nq** [comment]
When I was talking to Volca, this was one of the debatable changes that I listed (debatable as in uncertain if we needed it). It makes it a little easier to find the config on Linux, since it isn't in the yuzu data directory. This would also make it easier to replace the user's config file with one someone else suggests, but I wonder if that gets superseded by the plan to package recommended configs with yuzu.

**Comment by VolcaEM** [comment]
Rebased; is this fine now?

**Comment by VolcaEM** [comment]
This PR may need to be updated if #4408 gets merged first

**Comment by ogniK5377** [comment]
Hi, this is not how you rebase a pull request. You need to run `git rebase origin/master`, not git pull. Please revert your git pull and rebase, thanks!

**Comment by VolcaEM** [comment]
Closing because I'm unable to rebase, now and in the near future (I don't have a PC with me)
If anyone wants to open a new PR "in the meantime", please feel free to do so

**Pull request closed without merging**